src/coingecko/Coingecko.ts

```diff
@@ -21,9 +21,19 @@ export type CoinGeckoPrice = {
   price: number;
 };
 
+type PriceCache = {
```

Comment:
I think this is a fair observation.

To give a bit more context about what I see as the roadmap for the "price lookups" feature:
 - Implement a generic "PriceClient" class that acts as a client-facing implementation to query a price from.
 - During initialisation, the PriceClient instance has _n_ price feed adapters added, with a priority specified for each. For example:
   1. Across API
   2. CoinGecko Free
   2. CoinGecko Pro
   2. CoinMarketCap (??)
   2. On-chain DEX (??)
 - When the caller requests a price from the PriceClient, it looks up internally in its in-memory cache.
   1. For a cache hit, it returns the price immediately.
   2. For a cache miss, it performs a price lookup request incrementally down its ordered list until it gets a newer response than it currently has.

The advantages of this approach, as I see them:
 - Allows PriceClient users (front end, v2 bots) to preferentially leverage the Across API endpoint.
 - In the event of a failed lookup to any price feed, the existence of fallback price feeds should allow a price resolution without blocking the bots.
 - The existence of a preferential ordering in lookups allows users to minimise their reliance on paid endpoints like CoinGecko Pro, and to only fall back to them when they are really needed (i.e. price feed of last resort).

Reply:
@pxrl I think the `PriceClient` design with built in caching is a great idea, where the cache length can be confirgurable.

Reply:
Going to mark this one resolved since the description has been migrated over to Linear: https://linear.app/uma/issue/ACX-264/improve-robustness-against-coingecko-rate-limiting